api/src/main/java/org/eclipse/microprofile/openapi/models/media/Schema.java

```diff
@@ -1949,4 +1949,43 @@ default Schema contentSchema(Schema contentSchema) {
         return this;
     }
 
+    /**
+     * Returns whether this Schema is a boolean schema.
+     * <p>
+     * If this property is not {@code null}, then all other properties are ignored and the schema will be represented by
+     * a boolean {@code true} or {@code false} value.
+     *
+     * @return the boolean value of this schema, or {@code null} if it is not a boolean schema
+     * @since 4.0
+     */
+    Boolean getBooleanSchema();
+
+    /**
+     * Sets this schema to a boolean value.
+     * <p>
+     * If this property is not {@code null}, then all other properties are ignored and the schema will be represented by
+     * a boolean {@code true} or {@code false} value.
+     *
+     * @param booleanSchema
+     *            the boolean value of this schema, or {@code null} if it is not a boolean schema
+     * @since 4.0
+     */
+    void setBooleanSchema(Boolean booleanSchema);
```

Comment:
_Implementation note_: having `setBooleanSchema` made it near-impossible to implement `Schema` as a thin wrapper around a Jackson representation of some JSON.

All the other setters merely change properties on the root object, but to implement this setter you have to change the root object itself. In Jackson, this means setting a new object on the parent and Jackson also doesn't allow you to traverse up a JSON tree (parents know about their children, but children don't know about their parents).

The other way of implementing this would be to have factory methods for an immutable boolean schema on `OASFactory`, but that would be very different than the way the rest of the MP OpenAPI model is designed.

There were other issues with implementing it as a wrapper around Jackson, e.g. calling code may assume that two calls to `getAllOf()` will return the same `Schema` instances, so I gave up on that approach.